cmd/server/pactasrv/pacta_version.go

```diff
@@ -2,48 +2,164 @@ package pactasrv
+	// TODO(#12) Implement Authorization
+	pv, err := s.DB.PACTAVersion(s.DB.NoTxn(ctx), pacta.PACTAVersionID(request.Id))
+	if err != nil {
+		return nil, errorInternal(err)
+	}
+	return pactaVersionToOAPI(pv)
 }
 
 // Returns all versions of the PACTA model
 // (GET /pacta-versions)
 func (s *Server) ListPactaVersions(ctx context.Context, request api.ListPactaVersionsRequestObject) (api.ListPactaVersionsResponseObject, error) {
-	return nil, fmt.Errorf("not implemented")
+	pvs, err := s.listPactaVersions(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.ListPactaVersionsdefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.ListPactaVersions200JSONResponse(pvs), nil
+}
+
+func (s *Server) listPactaVersions(ctx context.Context, _request api.ListPactaVersionsRequestObject) ([]api.PactaVersion, error) {
+	// TODO(#12) Implement Authorization
+	pvs, err := s.DB.PACTAVersions(s.DB.NoTxn(ctx))
+	if err != nil {
+		return nil, errorInternal(err)
+	}
+	return dereference(mapAll(pvs, pactaVersionToOAPI))
 }
 
 // Creates a PACTA version
 // (POST /pacta-versions)
 func (s *Server) CreatePactaVersion(ctx context.Context, request api.CreatePactaVersionRequestObject) (api.CreatePactaVersionResponseObject, error) {
-	// TODO(grady) Authz
-	_, err := s.DB.CreatePACTAVersion(s.DB.NoTxn(ctx), &pacta.PACTAVersion{
-		Name:        request.Body.Name,
-		Description: request.Body.Description,
-		Digest:      request.Body.Digest,
-	})
+	err := s.createPactaVersion(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.CreatePactaVersiondefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.CreatePactaVersion200JSONResponse(emptySuccess), nil
+}
+
+func (s *Server) createPactaVersion(ctx context.Context, request api.CreatePactaVersionRequestObject) error {
+	// TODO(#12) Implement Authorization
+	pv, err := pactaVersionCreateToPACTA(request.Body)
 	if err != nil {
-		return nil, zap.Error(ctx, "failed to create PACTA version", zap.Error(err))
+		return errorBadRequest("body", err.Error())
 	}
-	return nil, nil
+	if _, err := s.DB.CreatePACTAVersion(s.DB.NoTxn(ctx), pv); err != nil {
+		return errorInternal(err)
+	}
+	return nil
 }
 
 // Updates a PACTA version
 // (PATCH /pacta-version/{id})
 func (s *Server) UpdatePactaVersion(ctx context.Context, request api.UpdatePactaVersionRequestObject) (api.UpdatePactaVersionResponseObject, error) {
-	return nil, fmt.Errorf("not implemented")
+	err := s.updatePactaVersion(ctx, request)
+	if err != nil {
+		e := errToAPIError(err)
+		return api.UpdatePactaVersiondefaultJSONResponse{
+			Body:       e,
+			StatusCode: int(e.Code),
+		}, nil
+	}
+	return api.UpdatePactaVersion200JSONResponse(emptySuccess), nil
+}
+
+func (s *Server) updatePactaVersion(ctx context.Context, request api.UpdatePactaVersionRequestObject) error {
```

Comment:
As an aside, I'm glad we're finally able to have the PATCH-style update semantics we never quite got right in GraphQL, this code is really easy to read + understand

Reply:
Totally agreed. I feel like in this way (and many others) this project is reaping benefits of the paradigms we didn't enjoy previously.